### PR TITLE
scattergl selection fixes

### PR DESCRIPTION
**Pull request by etpinard**
fixes https://github.com/plotly/plotly.js/issues/3740, https://github.com/plotly/plotly.js/issues/2958

before (context in #3740): https://codepen.io/Convo/pen/GLZYdw
after: https://codepen.io/etpinard/pen/xemRqV

@plotly/plotly_js see commit msg in https://github.com/plotly/plotly.js/commit/81367f13e3ee5bb6abfeda5aa12d36c182fac466 for more info.

### CODE REVIEW

File: src/traces/scattergl/index.js [hunk 1]
```diff
@@ -946,7 +938,7 @@ function styleTextSelection(cd) {
     var opts = Lib.extendFlat({}, baseOpts);
     var i, j;
 
-    if(els && unels) {
+    if(els.length || unels.length) {
```
Comment (archmoj) [inline src/traces/scattergl/index.js]: @etpinard Could you elaborate why .AND. condition is replaced by .OR. here?

Comment (etpinard) [reply src/traces/scattergl/index.js]: See commit message in https://github.com/plotly/plotly.js/pull/3810/commits/81367f13e3ee5bb6abfeda5aa12d36c182fac466

**Comment by archmoj** [comment]
Excellently done.
:dancer:

**Comment by etpinard** [comment]
Ok let's start merging things for 1.48.0

**Pull request merged**